manifests/agent/config.pp

```diff
@@ -1,91 +1,62 @@
 class puppet::agent::config {
 
-  include puppet
-  include puppet::agent
-  include puppet::params
-
   Ini_setting {
-    path    => $puppet::params::puppet_conf,
+    path    => $puppet::conf,
+    section => 'agent',
     ensure  => 'present',
   }
 
-  ini_setting { 'server':
-    section => 'main',
-    setting => 'server',
-    value   => $puppet::agent::server,
+  if $puppet::agent::ca_server {
+    $real_ca_server = $puppet::agent::ca_server
+  } else {
+    $real_ca_server = $puppet::agent::server
   }
 
-  ini_setting { 'ca_server':
-    section => 'main',
-    setting => 'ca_server',
-    value   => $puppet::agent::real_ca_server,
+  if $puppet::agent::report_server {
+    $real_report_server = $puppet::agent::report_server
+  } else {
+    $real_report_server = $puppet::agent::server
   }
 
-  ini_setting { 'report_server':
-    section => 'main',
-    setting => 'report_server',
-    value   => $puppet::agent::real_report_server,
-  }
-
-  if $puppet::agent::report_format {
-    ini_setting { 'report_format':
+  ini_setting {
+    'server':
       section => 'main',
-      setting => 'report_format',
-      value   => $puppet::agent::report_format,
-    }
-  }
-
-  ini_setting { 'pluginsync':
```

Comment:
FWIW, I prefer to have the resources broken out individually than lumped together.